tests/Rules/AuthorizedTest.php

```diff
@@ -7,79 +7,73 @@
 use Illuminate\Support\Facades\Gate;
 use Spatie\ValidationRules\Tests\TestCase;
 use Spatie\ValidationRules\Rules\Authorized;
+use Spatie\ValidationRules\Tests\TestClasses\Models\TestModel;
+use Spatie\ValidationRules\Tests\TestClasses\Policies\TestModelPolicy;
 
 class AuthorizedTest extends TestCase
 {
     public function setUp()
     {
         parent::setUp();
 
-        $this->actingAs(factory(User::class)->create());
+        Gate::policy(TestModel::class, TestModelPolicy::class);
     }
 
     /** @test */
     public function it_will_return_true_if_the_gate_returns_true_for_the_given_ability_name()
     {
-        Gate::define('abilityName', function () {
-            return true;
-        });
+        $rule = new Authorized('edit', TestModel::class);
 
-        $rule = new Authorized('abilityName');
+        $user = factory(User::class)->create(['id' => 1]);
+        factory(TestModel::class)->create([
+            'id' => 1,
+            'user_id' => $user->id,
+        ]);
 
-        $this->assertTrue($rule->passes('attribute', 'value'));
-    }
-
-    /** @test */
-    public function it_will_return_false_if_the_gate_returns_true_for_the_given_ability_name()
-    {
-        Gate::define('otherAbilityName', function () {
-            return false;
-        });
-
-        $rule = new Authorized('otherAbilityName');
+        $this->actingAs($user);
 
-        $this->assertFalse($rule->passes('attribute', 'value'));
+        $this->assertTrue($rule->passes('attribute', '1'));
     }
 
     /** @test */
-    public function it_will_return_false_if_there_is_noone_logged_in()
+    public function it_will_return_false_if_noone_is_logged_in()
     {
-        Auth::logout();
-
-        Gate::define('noUserAbilityName', function () {
-            return true;
-        });
+        $rule = new Authorized('edit', TestModel::class);
 
-        $rule = new Authorized('noUserAbilityName');
+        $user = factory(User::class)->create(['id' => 1]);
+        factory(TestModel::class)->create([
+            'id' => 1,
+            'user_id' => $user->id,
+        ]);
 
-        $this->assertFalse($rule->passes('attribute', 'value'));
+        $this->assertFalse($rule->passes('attribute', '1'));
     }
 
     /** @test */
-    public function it_can_accept_an_argument()
+    public function it_will_return_false_if_the_model_is_not_found()
     {
-        Gate::define('argumentAbilityName', function (User $user, $argument) {
-            return $argument === true;
-        });
+        $rule = new Authorized('edit', TestModel::class);
 
-        $rule = new Authorized('argumentAbilityName', true);
-        $this->assertTrue($rule->passes('attribute', 'value'));
+        $user = factory(User::class)->create(['id' => 1]);
+        factory(TestModel::class)->create([
```

Comment:
Maybe we can use `TestModel::create()`? That'll look cleaner - no added benefits from the factory.

Reply:
True, nuke the factory! 💣